src/telemetrify/main/test/common.act

```diff
@@ -0,0 +1,381 @@
+        except Exception as ex:
+            report_result(False, ex)
+
+    def test2_2():
+        try:
+            e.expect(["DO_STOP", "DO_START", "RUNNING"], test2_3)
+
+            DelegateMiddleManager_try_restart_inner(m)
+        except Exception as ex:
+            report_result(False, ex)
+
+    def test2_3():
+        try:
+            e.expect(["DO_STOP", "STOPPED"], test2_4)
+
+            testing.assertTrue(DelegateMiddleManager_set_target_state(m, STATE_STOPPED), "Invalid target state")
+        except Exception as ex:
+            report_result(False, ex)
+
+    def test2_4():
+        try:
+            e.expect(["STOPPED"], None)
+
+            DelegateMiddleManager_try_restart_inner(m) # Expecting NOOP
+            DelegateMiddleManager_set_target_state(m, STATE_STOPPED) # Verify STOPPED without any DO_STOP from the above restart
+        except Exception as ex:
+            report_result(False, ex)
+
+    def test3():
+        try:
+            _init(True)
+
+            add_delegate(key0)
+
+            e.expect(["DO_START", "RUNNING", "0-0_RUNNING"], test3_1)
+
+            testing.assertTrue(DelegateMiddleManager_set_target_state(m, STATE_RUNNING), "Invalid target state")
+        except Exception as ex:
+            report_result(False, ex)
+
+    def test3_1():
+        try:
+            e.expect(["0-0_CLOSED"], None)
+
+            DelegateMiddleManager_remove_delegate(m, key0)
+        except Exception as ex:
+            report_result(False, ex)
+
+    def test4():
+        try:
+            _init(True)
+
+            add_delegate(key0)
+
+            e.expect(["DO_START", "RUNNING", "0-0_RUNNING"], test4_1)
+
+            testing.assertTrue(DelegateMiddleManager_set_target_state(m, STATE_RUNNING), "Invalid target state")
+        except Exception as ex:
+            report_result(False, ex)
+
+    def test4_1():
+        try:
+            e.expect(["0-0_CLOSED", "0-1_RUNNING"], None)
+
+            DelegateMiddleManager_remove_delegate(m, key0)
+            add_delegate(key0)
+        except Exception as ex:
+            report_result(False, ex)
+
+def _test_async_dummy(report_result, log_handler: logging.Handler) -> None:
+    report_result(True, None)
+
+def _test_env_dummy(report_result, env, log_handler: logging.Handler) -> None:
+    report_result(True, None)
+
+__unit_tests: dict[str, testing.UnitTest] = {
+    "_test_dummy": testing.UnitTest(lambda: None, "dummy", "dummy"),
+}
+
+__sync_actor_tests: dict[str, testing.SyncActorTest] = {
+    "_test_sync_dummy": testing.SyncActorTest(lambda lh: None, "sync_dummy", "sync_dummy"),
+}
+
+__async_actor_tests: dict[str, testing.AsyncActorTest] = {
+    #"_test_async_dummy": testing.AsyncActorTest(_test_async_dummy, "async_dummy", "async_dummy"),
+    "_test_delegate_worker_1": testing.AsyncActorTest(lambda r, lh: DelegateWorkerTester(r, lh).test1(), "DelegateWorker1", "DelegateWorker state transitions #1"),
```

Comment:
Heh, just FYI, test discovery is implemented in actonc tip and it won't understand that it should do lambda here, you would need to instead define a explicit function, I gues something like:

```python
def _test_delegate_worker_test_1(r, lh):
    DelegateWorkerTester(r, lh).test1()
```

so maybe better to prepare the code for that already now? :)